### PR TITLE
Add uname data on different keys

### PR DESCRIPTION
**Pull request by fortizc**
The output of `pvcontrol devmeta ls | jq` is something like:
```json
{
  "storage": {
    "total": 15610982400,
    "free": 15318548480,
    "reserved": 780549120,
    "real_free": 14537999360
  },
  "pantavisor.version": "015-rc4-32-gfe4e999-220622-136fbe4",
  "pantavisor.uname": {
    "kernel.name": "Linux",
    "kernel.release": "4.19.127-v8+",
    "kernel.version": "#1 SMP PREEMPT Tue Jun 14 19:59:42 UTC 2022",
    "node.name": "(none)",
    "machine": "aarch64"
  },
  "pantavisor.revision": "locals/pbox-1655906948",
  "pantavisor.mode": "local",
  "pantavisor.dtmodel": "Raspberry Pi 3 Model B Plus Rev 1.3",
  "pantavisor.arch": "aarch64/64/EL",
  "pantahub.state": "update",
  "pantahub.online": "0",
  "pantahub.claimed": "1",
  "pantahub.address": "163.172.132.38:443",
  "interfaces": {
    "lo.ipv4": [
      "127.0.0.1"
    ],
    "enxb827eb52df46.ipv4": [
      "192.168.1.89"
    ],
    "lxcbr0.ipv4": [
      "10.0.3.1"
    ],
    "lo.ipv6": [
      "::1"
    ],
    "enxb827eb52df46.ipv6": [
      "fe80::54f3:4de4:569a:b0e3%enxb827eb52df46"
    ]
  }
}
```

### CODE REVIEW

File: metadata.c [hunk 1]
```diff
@@ -152,16 +152,48 @@ static int pv_devmeta_uname(struct pv_devmeta_read *pv_devmeta_read)
 	if (pv_devmeta_buf_check(pv_devmeta_read))
 		return -1;
 
+	char out[2048] = {0};
 	char err[2048] = {0};
-	tsh_run_output("uname -a", 2, buf, buflen, err, 2048);
+	tsh_run_output("uname -a", 2, out, 2048, err, 2048);
```
Comment (asac) [inline metadata.c:157]: why 2048? is that a known upper bound?

File: metadata.c [hunk 2]
```diff
@@ -152,16 +152,48 @@ static int pv_devmeta_uname(struct pv_devmeta_read *pv_devmeta_read)
...
 
-	if (strnlen(buf, buflen) == buflen) {
+  if (strnlen(out, 2048) == 0) {
```
Comment (asac) [inline metadata.c:159]: indent change

File: metadata.c [hunk 3]
```diff
@@ -152,16 +152,48 @@ static int pv_devmeta_uname(struct pv_devmeta_read *pv_devmeta_read)
...
+	++kversion;
+
+	char *os = strrchr(kversion, ' ');
+	*os = '\0';
+	++os;
+
+	char *arch = strrchr(kversion, ' ');
+	*arch = '\0';
+	++arch;
+
```
Comment (asac) [inline metadata.c:187]: is splitting the uname -a output in this order always correct?

i would suggest we rather use the posix c function uname() ... see man 2 uname...

**Review by asac — CHANGES_REQUESTED** [review]

File: metadata.c (hunk 3 above)
Comment (fortizc) [reply metadata.c]: Fixed in a39869406c384d6e2714349ab06e21bc4afdc311

File: metadata.c [hunk 4]
```diff
@@ -152,17 +152,24 @@ static int pv_devmeta_uname(struct pv_devmeta_read *pv_devmeta_read)
...
 		return -1;
 
-	char err[2048] = {0};
-	tsh_run_output("uname -a", 2, buf, buflen, err, 2048);
+	struct utsname data = {0};
+	int err = uname(&data);
 
-	if (strnlen(buf, buflen) == buflen) {
+	if (err) {
 		memset(buf, 0, buflen);
```
Comment (anibalportero) [inline metadata.c:159]: maybe add here pv_log(ERROR, "uname failed: %s", path, strerror(errno));

**Pull request merged**